### ISSUE TITLE
a content type that supports internal and external links

### ISSUE THREAD

**Issue by GillesKnockaert** [sulu/sulu#4966]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| Sulu Version | any
| Browser Version | any

#### Actual Behavior

currently you can add a content type "URL" -> this is used for external urls only.
you also have the content type single page selection -> here you can select a internal page to link to.

#### Expected Behavior

I would like have block that allows the user to create a button. but at this point i need to create 2 blocks, one for internal links and one for external links. 

Another option i have is just put both content types in the block and let a switch decide if i wan't to use the internal or external link ( or a work around in twig,...)

basically every option here is not that good or not user friendly.

#### Possible Solutions

A solution for this would be add a content type that makes both these things possible or edit the url content type so that you can set a param as internal link. ( mailto en tel would also be nice to have here).

If you then select internal link as param, you get the same affect as if you chose internal link as content type.

![image](https://user-images.githubusercontent.com/7395323/71909010-5bd46b00-316f-11ea-9662-2a72766db709.png)

**Comment by danrot** [comment]
I know it is not exactly the same, but at least partly related to #3873. And both of these fields are substantially different, so I don't know if there is an easy and usable way to combine them... When reading the issue title, I even thought about it the other way round: Instead of using blocks (I think that's what you are suggesting here) having a content type acting more like the `teaser_selection`. So a separate box like e.g. the `page_selection`, just that the button in to add stuff is a dropdown, and you can choose between an internal or external link, whereby both options are opening an overlay.

However, I would have to further discuss that, since I am not sure if that is really what we want :confused:

**Comment by GillesKnockaert** [comment]
well not sure on how to implement it but at this point there isn't a user friendly way to let a user create a link were he can chose between an internal or an external link. It would be nice if there was a feature for this ;)

**Comment by danrot** [comment]
I've talked to the other developers, and we have already been something like that for a project. It didn't include external links (just mixing links to pages, articles, media, ...), but it shouldn't be too big of a deal to include something like this. However, I think we currently have tasks with a higher priority right now... But if you are willing to contribute a feature like this I'd be happy to accept a PR :slightly_smiling_face:

**Comment by niklasnatter** [comment]
Related to https://github.com/sulu/sulu/issues/5790

**Comment by niklasnatter** [comment]
This was fixed implemented in #6134 and will be included in the upcoming Sulu 2.4 release 🎉 
We plan to publish the first stable Sulu 2.4 release in the end of 2021.
